b2share/modules/deposit/api.py

```diff
@@ -224,6 +226,25 @@ def submit(self, commit=True):
         if commit:
             self.commit()
 
+    def validate_before_publishing(self):
+        """Common sense validations before submitting. These would be also
```

Comment:
There will probably be other messages considered cryptic. Can't we catch the exception raised by the jsonschema library and raise our own instead? It would avoid reimplementing every validation ourselves.